cmd/kubectl/kubectl.go

```diff
@@ -119,34 +118,32 @@ func NewK0sKubectlCmd() *cobra.Command {
 	return cmd
 }
 
-func fallbackToK0sKubeconfig(args []string) error {
-	for _, arg := range args {
-		if arg == "--" {
-			// no more options
-			break
-		}
-		if arg == "--kubeconfig" || strings.HasPrefix(arg, "--kubeconfig=") {
-			// kubeconfig set via args, no need to check if k0s kubeconfig is readable
-			return nil
-		}
+func fallbackToK0sKubeconfig(cmd *cobra.Command) error {
```

Comment:
I think we don't need to set the flag if the environment variable is present. Kubectl will do this on its own.

An alternative proposal:

```go
func fallbackToK0sKubeconfig(cmd *cobra.Command) error {
	kubeconfigFlag := cmd.Flags().Lookup("kubeconfig")
	if kubeconfigFlag == nil {
		return fmt.Errorf("kubeconfig flag not found")
	}

	if kubeconfigFlag.Changed {
		return nil
	}

	if os.Getenv("KUBECONFIG") != "" {
		return nil
	}

	kubeconfig := config.GetCmdOpts().K0sVars.AdminKubeConfigPath

	// verify that k0s's kubeconfig is readable before setting the flag
	if _, err := os.Stat(kubeconfig); err != nil {
		return fmt.Errorf("cannot stat k0s kubeconfig, is the server running?: %w", err)
	}

	if err := kubeconfigFlag.Value.Set(kubeconfig); err != nil {
		return fmt.Errorf("failed to set kubeconfig flag: %w", err)
	}
	return nil
}
```